### PR TITLE
fix: boards: sam0: uart: remove collision_detect parsing

### PR DESCRIPTION
**Pull request by sm3vhf**
Removing collision detect property from devicetree parsing, no
collision_detect parameter included in device tree for any platform

### CODE REVIEW

**Comment by nandojve** [comment]
Hi @arvinf ,

Would like you comment about this change ?

Original PR: #34715

File: drivers/serial/uart_sam0.c [hunk 1]
```diff
@@ -1257,7 +1254,6 @@ static const struct uart_sam0_dev_cfg uart_sam0_config_##n = {		\
 	.mclk_mask = BIT(DT_INST_CLOCKS_CELL_BY_NAME(n, mclk, bit)),	\
 	.gclk_core_id = DT_INST_CLOCKS_CELL_BY_NAME(n, gclk, periph_ch),\
 	.pads = UART_SAM0_SERCOM_PADS(n),				\
-	.collision_detect = UART_SAM0_SERCOM_PADS(n),			\
```
Comment (nandojve) [inline drivers/serial/uart_sam0.c:1260]: Why not just fix instead remove it?

```diff
-.collision_detect = UART_SAM0_SERCOM_PADS(n), \
+.collision_detect = DT_INST_PROP(n, collision_detect), \
```

**Review by nandojve — CHANGES_REQUESTED** [review]

**Comment by sm3vhf** [comment]
@nandojve sure, i'll try!! I'm quite new to the process though, so not sure everything with the pull request will be correct all the way, but i'll try ;)

**Comment by nandojve** [comment]
Hi @sm3vhf ,
> @nandojve sure, i'll try!! I'm quite new to the process though, so not sure everything with the pull request will be correct all the way, but i'll try ;)

No problem, you should use correct property which is defined here:
https://github.com/zephyrproject-rtos/zephyr/blob/da041942c286c985d34543b676e560c1e2a6c77a/dts/bindings/serial/atmel%2Csam0-uart.yaml#L32

Please, update `DT_INST_PROP` to `DT_INST_PROP_OR` with default value of `0 (false)`

```diff
-.collision_detect = DT_INST_PROP(n, collision_detect), \
+.collision_detect = DT_INST_PROP_OR(n, collision_detection, 0), \
```

**Review by nandojve — DISMISSED** [review]

File: drivers/serial/uart_sam0.c [hunk 2]
```diff
@@ -1246,7 +1246,7 @@ static void uart_sam0_irq_config_##n(const struct device *dev)		\
 	(DT_INST_PROP(n, txpo) << SERCOM_USART_CTRLA_TXPO_Pos)
 
 #define UART_SAM0_SERCOM_COLLISION_DETECT(n) \
-	(DT_INST_PROP(n, collision_detect))
+	(DT_INST_PROP_OR(n, collision_detection, 0))
```
Comment (stephanosio) [inline drivers/serial/uart_sam0.c:1249]: `collision_detect` is `bool`, so it should be `false`.

**Review by stephanosio — CHANGES_REQUESTED** [review]

**Review by stephanosio — DISMISSED** [review]

**Review by nandojve — DISMISSED** [review]

**Comment by sm3vhf** [comment]
What is causing the failing tests? Do I need to make any more changes for those??

**Review by nandojve — DISMISSED** [review]
I think CI was complaining about other stuff not related to your PR.

**Review by stephanosio — DISMISSED** [review]

**Comment by sm3vhf** [comment]
@nandojve Thank you for the help!! 
Is there something I am missing to do? Is the merge expected to merge automatically, or is there something more left for me to do??

**Review by stephanosio — APPROVED** [review]

**Review by nandojve — APPROVED** [review]

**Pull request merged**